client/extensions/woocommerce/app/dashboard/required-plugins-install-view.js

```diff
@@ -0,0 +1,144 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import { find } from 'lodash';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { installPlugin } from 'state/plugins/installed/actions';
+import { fetchPluginData } from 'state/plugins/wporg/actions';
+import { getPlugin } from 'state/plugins/wporg/selectors';
+import { getPlugins } from 'state/plugins/installed/selectors';
+import { getSelectedSiteWithFallback } from 'woocommerce/state/sites/selectors';
+import ProgressBar from 'components/progress-bar';
+import QueryPluginKeys from 'components/data/query-plugin-keys';
+import QueryJetpackPlugins from 'components/data/query-jetpack-plugins';
+import SetupHeader from './setup-header';
+import { setFinishedInstallOfRequiredPlugins } from 'woocommerce/state/sites/setup-choices/actions';
+
+const requiredPlugins = [
+	'woocommerce',
+	// 'wc-api-dev',
+	'woocommerce-gateway-stripe',
+	'woocommerce-services',
+];
+
+class RequiredPluginsInstallView extends Component {
+	static propTypes = {
+		fetchPluginData: PropTypes.func.isRequired,
+		installPlugin: PropTypes.func.isRequired,
+		plugins: PropTypes.array,
+		setFinishedInstallOfRequiredPlugins: PropTypes.func.isRequired,
+		site: PropTypes.shape( {
+			ID: PropTypes.number.isRequired,
+		} ),
+		wpOrg: PropTypes.array,
+	};
+
+	constructor( props ) {
+		super( props );
+		this.state = {
+			installingPlugin: null,
+			progress: 0,
+		};
+	}
+
+	componentDidMount = () => {
+		const { plugins } = this.props;
+
+		if ( plugins && plugins.length ) {
```

Comment:
We should probably make sure the site is completely loaded before kicking off the install too.